packages/dynamite/dynamite_end_to_end_test/test/enum_test.dart

```diff
@@ -22,4 +25,22 @@ void main() {
       expect(deserialized, EnumString.$default);
     });
   });
+
+  group('int', () {
+    test('serialize', () {
+      final serialized = serializers.serialize(
+        EnumInt.$2,
+        specifiedType: const FullType(EnumInt),
+      );
+      expect(serialized, '2');
```

Comment:
why do we expect a string value and not a plain `2`?

Reply:
EnumClass can only take Strings as the name and this means it can only be serialized as such. We need to chain it with a `String->T where Enum<T>` serializer :thinking: Since I'm a noob, can you tell me how to do that? :see_no_evil:

Reply:
Or we change EnumClass to accept any type which I would prefer as we won't need the workaround for enums that are not strings.

Reply:
But that would be a breaking change for built_value so might not get accepted so easily.

Reply:
I'm curious whether this is even intended for openapi :thinking: 
And you already guessed that breaking changes in built_value are really hard to land upstream as this means all google internal projects must adapt and this can be a really slow process.

Reply:
You mean if integer enums are intended?

Reply:
Yes

Reply:
Yes the are. I previously also thought they were not allowed, but they are.

Reply:
https://json-schema.org/understanding-json-schema/reference/enum

Reply:
I think this is still a problem? I'll add a test that checks that the enum is correctly serialized in json but I'm pretty sure it will fail.